addons/mail/models/mail_message.py

```diff
@@ -868,22 +868,14 @@ def _message_reaction(self, content, action):
             self.env["mail.message.reaction"].create(create_values)
         if action == "remove" and reaction:
             reaction.unlink()
-        # format result
+        # format result to send to bus
         group_domain = [("message_id", "=", self.id), ("content", "=", content)]
-        count = self.env["mail.message.reaction"].search_count(group_domain)
-        group_command = "ADD" if count > 0 else "DELETE"
-        group_values = {
-            "content": content,
-            "count": count,
-            "personas": Store.many_ids(guest or partner, "ADD" if action == "add" else "DELETE"),
-            "message": Store.one_id(self),
-        }
-        # sudo: mail.guest - guest can send their own name when reacting
-        self._bus_send_store(
-            Store(self, {"reactions": [(group_command, group_values)]}).add(
-                guest.sudo() or partner, fields=["name", "write_date"]
-            )
-        )
+        reactions = self.env["mail.message.reaction"].search(group_domain)
+        group_command = "ADD" if len(reactions) > 0 else "DELETE"
+        reaction_group = Store.many(reactions, group_command)
+        if not reactions:
+            reaction_group = [(group_command, {"message": {"id": self.id}, "content": content})]
```

Comment:
Note:
For ideal API in business code, non-existence of records should be handled inside Store.many/_to_store.
I understand in this case, because of a we build reaction group, how that is not possible when deleting.